src/plugins/dashboard/public/application/hooks/use_dashboard_state_manager.ts

```diff
@@ -159,13 +160,22 @@ export const useDashboardStateManager = (
       stateManager.isNew()
     );
 
-    searchSession.setSearchSessionInfoProvider(
+    searchSession.setupStorage(
       createSessionRestorationDataProvider({
         data: dataPlugin,
         getDashboardTitle: () => dashboardTitle,
         getDashboardId: () => savedDashboard?.id || '',
         getAppState: () => stateManager.getAppState(),
-      })
+      }),
+      {
+        isDisabled: () =>
+          dashboardCapabilities.storeSearchSession
+            ? { disabled: false }
+            : {
+                disabled: true,
+                reasonText: noSearchSessionStorageCapabilityMessage,
```

Comment:
Could we please keep the reason text inside the plugin?
I think it's a too verbose API.

Reply:
current API doesn't limit disabling the UI for other reasons than not enough permissions. 
In case some other app has other app-specific use cases where they want to disable storing a session, they can still use this API by providing different explanation text.

For example, in a dashboard, I'd use this to disable storing a session when in edit mode.

Reply:
:+1: